app/src/main/java/com/example/firstapplication/UserAdapter.kt

```diff
@@ -34,22 +35,18 @@ class UserAdapter(
 
         val deleteUserButton: Button =  holder.deleteUserButton
         deleteUserButton.setOnClickListener {
-            remove(position)
+            remove(position, users[position].id)
         }
     }
 
     override fun getItemCount(): Int {
         return users.size
     }
 
-    fun add(user: User) {
-        users.add(user)
-        notifyItemInserted(users.lastIndex)
-    }
-
-    private fun remove(position: Int) {
+    private fun remove(position: Int, userId: String) {
         users.removeAt(position)
         notifyItemRemoved(position)
+        userListViewModel.remove(userId)
```

Comment:
skoro logikę usuwania z listy masz teraz po stronie viewmodelu, to linie 47 i 48 nie powinny być tutaj potrzebne